src/filesystem/filesystem_native.rs

```diff
@@ -135,7 +135,7 @@ impl super::filesystem_trait::Filesystem for Filesystem {
         path: impl ToString,
     ) -> Result<(), String> {
         let mut path = PathBuf::from(path.to_string());
-        let mut original_path = path.clone().to_string_lossy().to_string();
+        let original_path = path.clone().to_string_lossy().to_string();
```

Comment:
I think it's my code, but-
We should *fail* if we can't convert any `PathBuf` to a `String`.
[`PathBuff::to_string_lossy`](https://doc.rust-lang.org/std/path/struct.Path.html#method.to_string_lossy ) replaces all invalid characters with `U+FFFD` symbol - resulting in an invalid path, resulting in *path not found error when it's used to open a file or read directory.*

Reply:
Yeaaa I'm not sure what to do about that, but I can try and fix that later